spec/models/transformation_mapping_item_spec.rb

```diff
@@ -45,4 +46,60 @@
       expect(invalid_mapping_item.errors[:destination].first).to match("EMS type of destination cluster or cloud tenant must be in")
     end
   end
+  # end of destination cluster validation
+
+  context "datastore validation" do
+    let(:src_host) { FactoryBot.create(:host_vmware, :ems_cluster => vmware_cluster) }
+    let(:src)      { FactoryBot.create(:storage_vmware, :hosts => [src_host]) }
+
+    let(:ems) { FactoryBot.create(:ems_openstack) }
+    let(:disk) { FactoryBot.create(:disk) }
+    let(:cloud_tenant) { FactoryBot.create(:cloud_tenant_openstack, :ext_management_system => ems) }
+    let(:cloud_volume_openstack) { FactoryBot.create(:cloud_volume_openstack, :attachments => [disk], :cloud_tenant => cloud_tenant) }
+
+    before do
+      allow(vmware_cluster).to receive(:storages).and_return([src])
```

Comment:
Is this necessary?  Isn't the `EmsCluster#storages` method based on the relationships of `EmsCluster -> Host -> Storage`?

Reply:
This spec is going to change because the way I filter now has changed.

Reply:
Done.